docs/mesh/packing.md

```diff
@@ -115,3 +118,124 @@ MeshBlockVarFluxPack<T>
 ```
 which correspond to packs over meshblocks that contain just variables
 or contain variables and fluxes.
+
+## Registering MeshBlockPacks
```

Comment:
With this callback, btw, you only need to call this once after all packages are registered. Then the infrastructure has maximum flexibility to decided when and how new caches are allocated.

Reply:
It's important to note that this change will require a wider change to the code. Currently the tasking model is only kind-of coupled to the mesh, but not enough. We'll need to do a tighter coupling - we describe our task lists in a very granular way, and that would probably need to change, too.

Reply:
> As far as I'm concerned, the RequestMeshBlockPack callback shouldn't be concerned with partitioning at all. 

Hmm... I'm not sure how *not* to include that. I can imagine applications sometimes wanting to pack over `N` meshblocks and sometimes over `M`. I think that needs to be available somehow. However, I agree that it should be automatically synchronized with `TaskCollection` if possible. And that's probably doable. 

> Since we're doing explicit caching here rather than implicit caching

That's going to change. Once we move to implicit caching, things should be more workable.

> It's important to note that this change will require a wider change to the code. Currently the tasking model is only kind-of coupled to the mesh, but not enough. We'll need to do a tighter coupling - we describe our task lists in a very granular way, and that would probably need to change, too.

This probably requires a *lot* more thought. But I think you're right that's a good path forward.

Reply:
> Hmm... I'm not sure how not to include that.

It's not that the pack shouldn't be per-partition, it's just that the user shouldn't be manually doing the partitioning.

Reply:
Right, but the user should be doing the partitioning somewhere. Either at the pack level, or at the tasking level.

Reply:
I think this is very much in line with what we discussed during the call.
With the implicit caching  (and invalidating the cache, e.g., after a load balancing call) the TaskCollection would be the only* place left where task packing logic is required and self-consistently handled for full cycle.

\* plus the calls in the infrastructure (e.g., derived fields or EstimateTimestep), but I'm in favor of changing those interfaces in a way that the information required (about the packs or directly about the `MeshBlocks` if the application decides to define a Task at the MeshBlock level) is provided from within `TaskCollection` rather than extracted from logic in a container

Reply:
Yes, I think that's true.